includes/class-wpcom-legacy-redirector.php

```diff
@@ -385,17 +421,17 @@ public static function check_if_404( $url ) {
 		$response_code = '';
 		if ( is_array( $response ) ) {
 			$response_code = wp_remote_retrieve_response_code( $response );
-
 		}
 		return $response_code;
 	}
+
 	/**
 	 * Check if $redirect is a public Post.
 	 *
 	 * @param string $excerpt The Excerpt.
+	 * @return object The Post.
```

Comment:
Incorrect: this method returns a string.

Reply:
In the majority of cases it will return [WP_Post](https://codex.wordpress.org/Function_Reference/get_page_by_path). 

I will also add string to account private/null scenario's.